datasets/youtube.py

```diff
@@ -1,21 +1,59 @@
 import os
 import json
 import argparse
+import glob
 
-parser = argparse.ArgumentParser()
-parser.add_argument('--input-path', '-i', required = True)
-parser.add_argument('--output-path', '-o', required = True)
-parser.add_argument('--strip', nargs = '*', default = ['begin', 'end'])
-args = parser.parse_args()
 
-transcript = []
-for info_name in os.listdir(args.input_path):
-	if not info_name.endswith('.json'):
-		continue
-	
-	info_path = os.path.join(args.input_path, info_name)
-	transcript.extend(dict(audio_path = info_path.replace('.json', ''), **{k : v for k, v in t.items() if k not in args.strip} ) for t in json.load(open(info_path)).get('transcript', []))
+def main(args):
+	transcripts = []
+	for i, info_path in enumerate(glob.glob(os.path.join(args.input_path, f'*/*/*.json'))):
+		print(i)
 
-json.dump(transcript, open(args.output_path, 'w'), ensure_ascii = False, indent = 2, sort_keys = True)
+		j = json.load(open(info_path))
 
-print(args.output_path)
+		ref_char_count = sum([len(t.get('ref', '')) for t in j.get('transcript', [])])
+
+		json_transcripts = [dict(
```

Comment:
Semantics of `json_transcripts` is not clear, when there's an accumulator list `transcripts` (which is the one that will be dumped to json btw). In this codebase in all places where transcript-like dictionaries are handled, they are not specifically labeled "json_transcripts".

If `json_transcripts` is just a temp list to be accumulated in `transcripts`, just call it `transcripts_chunk` or even `transcripts_`